src/lib/agent/LedgerService.ts

```diff
@@ -0,0 +1,181 @@
+    logger.log('result', result);
+
+    return result;
+  }
+
+  async registerSchema(myDid: Did, schemaTemplate: SchemaTemplate): Promise<[SchemaId, Schema]> {
+    if (!this.poolHandle) {
+      throw new Error('Pool has not been initialized.');
+    }
+    const { name, attributes, version } = schemaTemplate;
+    const [schemaId, schema] = await this.indy.issuerCreateSchema(myDid, name, version, attributes);
+    logger.log('schemaId, schema', schemaId, schema);
+
+    const request = await this.indy.buildSchemaRequest(myDid, schema);
+    logger.log('request', request);
+
+    const requestWithTaa = await this.appendTaa(myDid, request);
+
+    const signedRequest = await this.wallet.signRequest(myDid, requestWithTaa);
+    logger.log('signedRequest', signedRequest);
+
+    const response = await this.indy.submitRequest(this.poolHandle, signedRequest);
+    logger.log('response', response);
+
+    return [schemaId, schema];
+  }
+
+  async getSchema(myDid: Did, schemaId: SchemaId) {
+    if (!this.poolHandle) {
+      throw new Error('Pool has not been initialized.');
+    }
+    const request = await this.indy.buildGetSchemaRequest(myDid, schemaId);
+    logger.log('request', request);
+
+    const response = await this.indy.submitRequest(this.poolHandle, request);
+    logger.log('response', response);
+
+    const result = await this.indy.parseGetSchemaResponse(response);
+    logger.log('Credential Schema from ledger: ', result);
+
+    return result;
+  }
+
+  async registerDefinition(myDid: Did, credentialDefinitionTemplate: CredDefTemplate): Promise<[CredDefId, CredDef]> {
+    if (!this.poolHandle) {
+      throw new Error('Pool has not been initialized.');
+    }
+    const { schema, tag, signatureType, config } = credentialDefinitionTemplate;
+
+    const [credDefId, credDef] = await this.wallet.createCredDef(myDid, schema, tag, signatureType, config);
+    const request = await this.indy.buildCredDefRequest(myDid, credDef);
+    logger.log('request', request);
+
+    const requestWithTaa = await this.appendTaa(myDid, request);
+
+    const signedRequest = await this.wallet.signRequest(myDid, requestWithTaa);
+    logger.log('signedRequest', signedRequest);
+
+    const response = await this.indy.submitRequest(this.poolHandle, signedRequest);
+    logger.log('response', response);
+
+    return [credDefId, credDef];
+  }
+
+  async getDefinition(myDid: Did, credDefId: CredDefId) {
+    if (!this.poolHandle) {
+      throw new Error('Pool has not been initialized.');
+    }
+    const request = await this.indy.buildGetCredDefRequest(myDid, credDefId);
+    logger.log('request', request);
+
+    const response = await this.indy.submitRequest(this.poolHandle, request);
+    logger.log('response', response);
+
+    const result = await this.indy.parseGetCredDefResponse(response);
+    logger.log('Credential Definition from ledger: ', result);
+
+    return result;
+  }
+
+  private async appendTaa(myDid: Did, request: LedgerRequest) {
```

Comment:
Maybe I don't understand it correctly, but do we need to do this for every request? I thought it was an accept once, then make requests as you like kinda agreement.

Reply:
Yes, it seems we need to add it to every request. VCX also mentions it here:
https://github.com/AbsaOSS/libvcx/blob/c3de77cb12da134f23b3434c534aebeb43119ea9/libvcx/src/api/vcx.rs#L419